core/trino-main/src/main/java/io/trino/operator/MultiChannelGroupByHash.java

```diff
@@ -80,12 +80,9 @@
     private int hashCapacity;
     private int maxFill;
     private int mask;
-    private long[] groupAddressByHash;
     private int[] groupIdsByHash;
```

Comment:
could you add a comment that `groupId` is also an address within `channelBuilders`? This works via a neat trick because rows are added incrementally, but it might not be obvious for a first time reader

Reply:
comment added, take a look if it makes sense to you.